src/recipe/recipe.repository.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { Recipe } from '@prisma/client';
+import { CreateRecipeRequest, UpdateRecipeRequest } from './dto';
+
+@Injectable()
+export class RecipeRepository {
+  constructor(private prisma: PrismaService) {}
+
+  async createRecipe(data: CreateRecipeRequest): Promise<Recipe> {
+    const recipe = await this.prisma.recipe.create({
+      data,
+    });
+
+    return recipe;
```

Comment:
you don't have to assign result to `recipe` and return it - you can return straight away:
```suggestion
    return this.prisma.recipe.create({
      data,
    });
```

Reply:
implementing above you can also remove `async` (`Promise` in `Promise<Recipe>` stays)